engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -82,18 +158,35 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
   when(WorkflowExecutionInProgressState) {
     case Event(BackendJobExecutionSucceededResponse(jobKey, callOutputs), stateData) =>
       log.info(s"Job ${jobKey.call.fullyQualifiedName} succeeded! Outputs: ${callOutputs.mkString("\n")}")
-      context.parent ! WorkflowExecutionSucceededResponse
-      goto(WorkflowExecutionSuccessfulState)
+      val newData = stateData.handleCompletedCall(jobKey, callOutputs, ExecutionStatus.Done) // addPersisting(jobKey, ExecutionStatus.Done)
+      if (newData.isWorkflowComplete) {
+        // Printing the final outputs, temporarily here until SingleWorkflowManagerActor is made in-sync with the shadow mode
```

Comment:
Could you at least move it out into a separate `generateAndPrintWorkflowOutputs` method?

It's just I find all this stuff is getting in the way of parsing the state machine.